### PR TITLE
Add of support

### PR DESCRIPTION
**Pull request by toleressea**
Add support for:
* anyOf
* oneOf (fixes existing)
* allOf

### CODE REVIEW

**Comment by toleressea** [comment]
This is definitely not the "soon" I was hoping for.

I cannot reproduce the issue I believed I saw before with the "oneOf" implementation @mehagar put in. Originally, I wrote tests and implemented all three (oneOf, anyOf, and allOf) before noticing @mehagar had added oneOf. I pulled it down, ran it against my tests and found it to not work. At that point, I should have added an issue to document what I had found, though now I can only acknowledge and apologize for incomplete testing of his implementation in the first place. At the very least, when I was creating my pull request the other day I should have re-tested the existing oneOf implementation.

tl;dr I erred up on multiple levels - I can find no fault with the oneOf implementation @mehagar put in. My own oneOf can be reverted and/or my other "Of" implementations could probably be made simpler by mimicking what @mehagar did originally. I'm sorry!

**Comment by penduin** [comment]
No problem @toleressea.  I've put "oneOf" validation back the way it was for now, just because, but if you or anybody else finds that it has a bug, I'll be most happy to take a patch and/or investigate a test case.  Thanks!
